pkg/generator_test.go

```diff
@@ -559,15 +559,285 @@ func NewExpecter(t mockConstructorTestingTNewExpecter) *Expecter {
+}
+
+// Variadic is a helper method to define mock.On call
+//   - ints ...int
+func (_e *Expecter_Expecter) Variadic(ints ...interface{}) *Expecter_Variadic_Call {
+	return &Expecter_Variadic_Call{Call: _e.mock.On("Variadic",
+		append([]interface{}{}, ints...)...)}
+}
+
+func (_c *Expecter_Variadic_Call) Run(run func(ints ...int)) *Expecter_Variadic_Call {
+	_c.Call.Run(func(args mock.Arguments) {
+		variadicArgs := make([]int, len(args)-0)
+		for i, a := range args[0:] {
+			if a != nil {
+				variadicArgs[i] = a.(int)
+			}
+		}
+		run(variadicArgs...)
+	})
+	return _c
+}
+
+func (_c *Expecter_Variadic_Call) Return(_a0 error) *Expecter_Variadic_Call {
+	_c.Call.Return(_a0)
+	return _c
+}
+
+func (_c *Expecter_Variadic_Call) RunAndReturn(run func(...int) error) *Expecter_Variadic_Call {
+	_c.Call.Return(run)
+	return _c
+}
+
+// VariadicMany provides a mock function with given fields: i, a, intfs
+func (_m *Expecter) VariadicMany(i int, a string, intfs ...interface{}) error {
+	var tmpRet mock.Arguments
+	if len(intfs) > 0 {
+		tmpRet = _m.Called(i, a, intfs)
+	} else {
+		tmpRet = _m.Called(i, a)
+	}
+
+	ret := tmpRet
+
+	var r0 error
+	if rf, ok := ret.Get(0).(func(int, string, ...interface{}) error); ok {
+		r0 = rf(i, a, intfs...)
+	} else {
+		r0 = ret.Error(0)
+	}
+
+	return r0
+}
+
+// Expecter_VariadicMany_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'VariadicMany'
+type Expecter_VariadicMany_Call struct {
+	*mock.Call
+}
+
+// VariadicMany is a helper method to define mock.On call
+//   - i int
+//   - a string
+//   - intfs ...interface{}
+func (_e *Expecter_Expecter) VariadicMany(i interface{}, a interface{}, intfs ...interface{}) *Expecter_VariadicMany_Call {
+	return &Expecter_VariadicMany_Call{Call: _e.mock.On("VariadicMany",
+		append([]interface{}{i, a}, intfs...)...)}
+}
+
+func (_c *Expecter_VariadicMany_Call) Run(run func(i int, a string, intfs ...interface{})) *Expecter_VariadicMany_Call {
+	_c.Call.Run(func(args mock.Arguments) {
+		variadicArgs := make([]interface{}, len(args)-2)
+		for i, a := range args[2:] {
+			if a != nil {
+				variadicArgs[i] = a.(interface{})
+			}
+		}
+		run(args[0].(int), args[1].(string), variadicArgs...)
+	})
+	return _c
+}
+
+func (_c *Expecter_VariadicMany_Call) Return(_a0 error) *Expecter_VariadicMany_Call {
+	_c.Call.Return(_a0)
+	return _c
+}
+
+func (_c *Expecter_VariadicMany_Call) RunAndReturn(run func(int, string, ...interface{}) error) *Expecter_VariadicMany_Call {
+	_c.Call.Return(run)
+	return _c
+}
+
+type mockConstructorTestingTNewExpecter interface {
+	mock.TestingT
+	Cleanup(func())
+}
+
+// NewExpecter creates a new instance of Expecter. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
+func NewExpecter(t mockConstructorTestingTNewExpecter) *Expecter {
+	mock := &Expecter{}
+	mock.Mock.Test(t)
+
+	t.Cleanup(func() { mock.AssertExpectations(t) })
+
+	return mock
+}
+`
+
```

Comment:
I am trying to move away from the tests asserting the exact string of the generated mock as we currently have to modify like 40 different tests whenever the mock generation code changes. Instead, the better way to do this is add the mock with the specified config into the `.mockery.yaml` file, and run tests off of that mock: https://github.com/vektra/mockery/blob/master/.mockery.yaml#L23

You'd need to add another section under the `configs` of `RequesterVariadic`, something like:
```yaml
- mockname: RequesterVariadicExpecterAndUnrollVariadic
  unroll-variadic: false
  with-expected: true
```

It should be good enough for now to commit the mock and run the test. I'm going to implement a change to CI/CD so that it generates the mocks _before_ running tests (it currently just generates them after).

Reply:
Adjusted to follow this testing pattern. I also cleaned up other tests to use the `require` interface and removed the incorrect messages.